adobe/react-spectrum
# [On-hold] RSP-1396 [Accessibility] Asterisk has ariaLabel and role presentation

Closes [RSP-1396](https://jira.corp.adobe.com/browse/RSP-1396) and [RSP-1360](https://jira.corp.adobe.com/browse/RSP-1360)

NOTE: this PR includes a commit to address the related issue for labelling and setting the role on Icon and UIIcon components [RSP-1360](https://jira.corp.adobe.com/browse/RSP-1360)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [RSP-1396](https://jira.corp.adobe.com/browse/RSP-1396) and [RSP-1360](https://jira.corp.adobe.com/browse/RSP-1360).

## 📝 Test Instructions:

1. Open http://localhost:9003/?path=/story/fieldlabel--required-styles
2. Verify spacing on FieldLabel with necessityLabel and Required Asterix icon
3. Using web inspector, verify that for the field labelled "Required label (required)", the neccessityLabel span has aria-hidden="true" and the related textfield input has aria-required="true"
4. Using web inspector, verify that for the field labelled "Optional label (options)", the neccessityLabel span does not have aria-hidden="true" and the related textfield input does not have aria-required="true"
5. Using web inspector, verify that for the field labelled "React *", the icon svg does not have an aria-label and has aria-hidden="true" and the related textfield input has aria-required="true"
6. Using web inspector, verify that for the field labelled "isRequired on child", the icon svg is rendered, does not have an aria-label and has aria-hidden="true" and the related textfield input has aria-required="true". This is the case where a single child with isRequired forces isRequired on the parent label.

## 🧢 Your Team:

Accessibility
